In [1]:
import numpy as np
import pandas as pd
import sidetable
import sqlite3
import warnings
warnings.filterwarnings('ignore')

In [24]:
stars = pd.read_csv("Stars.csv")
stars

Temperature              L          R    A_M  Color Spectral_Class  Type
0           3068       0.002400     0.1700  16.12    Red              M     0
1           3042       0.000500     0.1542  16.60    Red              M     0
2           2600       0.000300     0.1020  18.70    Red              M     0
3           2800       0.000200     0.1600  16.65    Red              M     0
4           1939       0.000138     0.1030  20.06    Red              M     0
..           ...            ...        ...    ...    ...            ...   ...
235        38940  374830.000000  1356.0000  -9.93   Blue              O     5
236        30839  834042.000000  1194.0000 -10.63   Blue              O     5
237         8829  537493.000000  1423.0000 -10.73  White              A     5
238         9235  404940.000000  1112.0000 -11.23  White              A     5
239        37882  294903.000000  1783.0000  -7.80   Blue              O     5

[240 rows x 7 columns]

In [15]:
stars["Color"].value_counts()

Color
Red                   112
Blue                   56
Blue-white             26
Blue White             10
yellow-white            8
White                   7
Blue white              4
white                   3
Yellowish White         3
yellowish               2
Whitish                 2
Orange                  2
White-Yellow            1
Pale yellow orange      1
Yellowish               1
Orange-Red              1
Blue-White              1
Name: count, dtype: int64

Red                112
Blue                56
Blue-White          41
White               10
Yellow-White         9
Yellowish-White      3
Yellowish            3
Whitish              2
Orange               2
Yellow-Orange        1
Orange-Red           1

In [25]:
replace_map = {'Red': 'Red', 
               'Blue White': 'Blue-White', 
               'White': 'White', 
               'Yellowish White': 'Yellow-White', 
               'Blue white': 'Blue-White',
               'Pale yellow orange': 'Yellow-Orange', 
               'Blue': 'Blue', 
               'Blue-white': 'Blue-White', 
               'Whitish': 'White',
               'yellow-white': 'Yellow-White', 
               'Orange': 'Orange', 
               'White-Yellow': 'Yellow-White', 
               'white': 'White', 
               'yellowish': 'Yellow',
               'Yellowish': 'Yellow', 
               'Orange-Red': 'Orange-Red', 
               'Blue-White': 'Blue-White'}

stars["Color"] = stars["Color"].replace(replace_map)
stars["Color"].value_counts()

Color
Red              112
Blue              56
Blue-White        41
White             12
Yellow-White      12
Yellow             3
Orange             2
Yellow-Orange      1
Orange-Red         1
Name: count, dtype: int64

In [26]:
replace_map = {
    0: 'Red Dwarf',
    1: 'Brown Dwarf',
    2: 'White Dwarf',
    3: 'Main Sequence',
    4: 'Super Giants',
    5: 'Hyper Giants'}
stars["Type"] = stars["Type"].replace(replace_map)
stars["Type"].value_counts()

Type
Red Dwarf        40
Brown Dwarf      40
White Dwarf      40
Main Sequence    40
Super Giants     40
Hyper Giants     40
Name: count, dtype: int64

In [27]:
stars.to_csv("Stars_clean_r.csv")

In [28]:
stars["L"] = stars["L"] * 3.828 * 10**26

In [29]:
stars["R"] = stars["R"] * 6.9551 * 10**8

In [30]:
stars.to_csv("Stars_clean_a.csv")